modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -133,9 +143,9 @@ class BaseConvolutionLayerImpl : public ConvolutionLayer
     virtual MatShape computeColRowShape(const MatShape &inpShape, const MatShape &outShape) const = 0;
     bool is1x1() const
     {
-        return (kernel.height == 1 && kernel.width == 1) &&
-        (stride.height == 1 && stride.width == 1) &&
-        (dilation.height == 1 && dilation.width == 1);
+        return (kernel_size.back() == 1 && kernel_size[kernel_size.size() - 2] == 1) &&
+               (strides.back() == 1 && strides[strides.size() - 2] == 1) &&
+               (dilations.back() == 1 && dilations[dilations.size() - 2] == 1);
```

Comment:
Due `1x1` method is called only int the default implementation for 2D convs only we can keep `cv::Size` checks because they will be actual.

Reply:
done